datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaTransportProvider.java

```diff
@@ -112,7 +112,7 @@ private ProducerRecord<byte[], byte[]> convertToProducerRecord(String topicName,
     Headers headers = null;
     if (event instanceof BrooklinEnvelope) {
       BrooklinEnvelope envelope = (BrooklinEnvelope) event;
-      headers = envelope.getHeaders();
+      headers = envelope.getHeaders() instanceof Headers ? (Headers) envelope.getHeaders() : null;
```

Comment:
Do we expect a non Kafka Headers object here? If no, is it worth throwing an exception or logging an error here?

Reply:
logging would be costly as it would do it for every record, I instead added an exception path which will fail in moving further if something other than kafka type is encountered.